contracts/interchain-token-service/src/state.rs

```diff
@@ -107,6 +140,89 @@ pub fn load_all_its_contracts(
         .collect::<Result<HashMap<_, _>, _>>()?)
 }
 
+pub fn save_token_info(
+    storage: &mut dyn Storage,
+    chain: ChainNameRaw,
+    token_id: TokenId,
+    is_origin_chain: bool,
+    track_balance: bool,
+) -> Result<(), Error> {
+    match TOKEN_INFO.may_load(storage, &(chain.clone(), token_id.clone()))? {
+        Some(_) if is_origin_chain => (),
+        Some(_) => return Err(Error::TokenNotDeployed { token_id, chain }),
+        None => {
+            let balance = TokenBalance {
+                balance: track_balance.then(Uint256::zero),
+                is_origin_chain,
+            };
+            TOKEN_INFO.save(storage, &(chain, token_id), &TokenInfo { balance })?;
+        }
+    }
+
+    Ok(())
+}
+
+pub fn update_token_info(
+    storage: &mut dyn Storage,
+    chain: ChainNameRaw,
+    token_id: TokenId,
+    amount: nonempty::Uint256,
+    is_deposit: bool,
+) -> Result<(), Error> {
+    let Some(mut token_config) =
+        TOKEN_INFO.may_load(storage, &(chain.clone(), token_id.clone()))?
+    else {
+        return Err(Error::TokenNotDeployed { token_id, chain });
+    };
```

Comment:
I feel like this logic should be separated out into it's own function: `is_token_deployed`

Reply:
actually nvm, this is fine here